src/util/nondet_ifthenelse.cpp

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************\
+
+Module: Nondeterministic if-then-else
+
+Author: Chris Smowton, chris@smowton.net
+
+\*******************************************************************/
+
+#include "nondet_ifthenelse.h"
+
+#include <sstream>
+
+#include <util/nondet_bool.h>
+#include <util/std_code.h>
+#include <util/std_types.h>
+#include <util/std_expr.h>
+#include <util/symbol_table.h>
+#include <util/source_location.h>
+
+// This will be unified with other similar fresh-symbol routines shortly
+static symbolt &new_tmp_symbol(
+  symbol_tablet &symbol_table,
+  const std::string &prefix,
+  const irep_idt &mode)
+{
+  static size_t temporary_counter=0;
+
+  auxiliary_symbolt new_symbol;
+  symbolt *symbol_ptr;
+
+  do
+  {
+    new_symbol.name="tmp_object_factory$"+std::to_string(++temporary_counter);
+    new_symbol.base_name=new_symbol.name;
+    new_symbol.mode=mode;
+  }
+  while(symbol_table.move(new_symbol, symbol_ptr));
+
+  return *symbol_ptr;
+}
+
+/*******************************************************************\
+
+Function: nondet_ifthenelset::begin_if
+
+  Inputs:
+
+ Outputs:
+
+ Purpose: Emits instructions and defines labels for the beginning of
+          a nondeterministic if-else diamond. Code is emitted to the
```

Comment:
Ah, true. I wrote this a long time ago when I thought that was how things were supposed to work here (since the Java frontend never generates high-level constructs, because bytecode). I'll take another look at this tomorrow.